crypto/src/immutable_bytes.rs

```diff
@@ -0,0 +1,90 @@
+#[cfg(not(feature = "std"))]
+use alloc::{boxed::Box, format, string::String, vec, vec::Vec};
+use core::mem;
+#[cfg(feature = "std")]
+use std::{boxed::Box, vec::Vec};
+
+use iroha_schema::IntoSchema;
+use parity_scale_codec::{Compact, Decode, Encode};
+use serde::{Deserialize, Serialize};
+
+/// Stores bytes that are not supposed to change during the runtime of the program
+///
+/// This is a more efficient than `Vec<u8>` because it does not have to store the capacity field
```

Comment:
This sounds like a compact pointer around `[u8]`, not an immutable. This pointer is good for keeping something in the storage compactly, but if we consider `Arc<[u8]>` as an immutable data structure, that may be more efficient because it has cheap copies.
I'd rename it to `CompactBytes` or something like that IMHO.

Reply:
Done

Reply:
Curiously, the same can be said about the `ConstString`: it doesn't have cheap copies, but its name can suggest this..

Reply:
I think these are not really suitable names for these data structures, and a bit confusing when reading them. But it's just my opinion, and I guess good documentation eliminates this problem at all.